pkg/rpc/rpcserver.go

```diff
@@ -68,8 +70,21 @@ func (s *Server) Start() error {
 		s.handleRequest(w, *r, isAdmin)
 	})
 
-	// Set up listener
-	l, err := net.Listen("tcp", "localhost:"+cfg.Get().RPC.Port)
+	// Generate self-signed certificate
+	tlsCert, err := cryptoUtils.GenerateTLSCertificate()
```

Comment:
Wouldn't it make sense to  try to read TLS cert and key file from config as first step? If TLS config is missing, then generate and use self-signed cert?

Reply:
We have a future TODO to provide certs from the voucher:

https://github.com/dusk-network/dusk-voucher/issues/2

Reply:
If the snippet below works for you we can handle this with one MR.

```go
certFile := cfg.Get().RPC.CertFile
keyFile := cfg.Get().RPC.KeyFile

keypair, err := tls.LoadX509KeyPair(certFile, keyFile)
if err != nil {
	return nil, err
}
```

Reply:
Done